content/blog/debugging-valgrind-gdb.md

```diff
@@ -0,0 +1,190 @@
+---
+title: "Debugging memory errors with valgrind and gdb"
+author: Rich FitzJohn
+date: 2021-06-10
+tags:
+ - dust
+ - R
+ - debugging
+---
+
+We've already written about debugging R packages with valgrind and gdb together [here](https://reside-ic.github.io/blog/debugging-and-fixing-crans-additional-checks-errors/). This post shows how to use them together so that you can use gdb to inspect the state of a program at the point where valgrind has noticed that there's an error.  The process is a bit weird, but does work!  This will likely only work on Linux (not windows or macOS).
```

Comment:
```suggestion
We've already written about debugging R packages with valgrind and gdb separately [here](https://reside-ic.github.io/blog/debugging-and-fixing-crans-additional-checks-errors/). This post shows how to use them together so that you can use gdb to inspect the state of a program at the point where valgrind has noticed that there's an error.  The process is a bit weird, but does work!  NB: This will likely only work on Linux (not windows or macOS).
```